crates/nargo/tests/test_data/9_conditional/src/main.nr

```diff
@@ -74,6 +74,10 @@ fn main(a: u32, mut c: [u32; 4], x: [u8; 5], result: pub [u8; 32]){
     constrain c[1] != 2;
     call_intrinsic(x, result);
 
+    //Test case for conditional with arrays from function parameters
+    let b = sort([1,2,3,4]);
+    constrain b[0] == 1;
```

Comment:
Perhaps we should test this sort function on an unsorted array

Reply:
This test case is failing without this PR with the array already sorted. This is because it believes it is created in the THEN branch, so it will re-order the array without predicate (i.e not taking into account the condition), which is the correct thing to do (if the array was created in the branch). As a result, this array will not be sorted and the test will fail.
If we use an unsorted array, we will probably get a sorted array and the test will not fail.
Long story short; we need a sorted array here.